custom_components/waste_collection_schedule/waste_collection_schedule/source/infeo_at.py

```diff
@@ -0,0 +1,89 @@
+import json
+import logging
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+from waste_collection_schedule.service.ICS import ICS
+
+_LOGGER = logging.getLogger(__name__)
+
+TITLE = "INFEO"
+DESCRIPTION = "Source for INFEO waste collection."
+URL = "https://www.infeo.at/"
+TEST_CASES = {"Bogenschütz": {"customer": "bogenschütz", "zone": "Dettenhausen"}}
+
+
+class Source:
+    def __init__(self, customer, zone):
+        self._customer = customer
+        self._zone = zone
+        self._ics = ICS()
+
+    def fetch(self):
+        baseUrl = f"https://services.infeo.at/awm/api/{self._customer}/wastecalendar"
+        issueUrl = "https://github.com/mampfes/hacs_waste_collection_schedule/issues/new"
+    
+        # get the available published calendar years
+        url = f"{baseUrl}/calendars/?showUnpublishedCalendars=false"
+        response = requests.request("GET", url)
+
+        # data validation
+        if(response.status_code != 200):
+            _LOGGER.error(f"problems during api calendar year access, please file an issue at {issueUrl} and mention @dm82m and add this: {response.text}")
+            return []
+            
+        response = response.json()
+        if len(response) <= 0:
+            _LOGGER.error(f"no calendars found, please file an issue at {issueUrl} and mention @dm82m")
+            return []
+
+        entries = []
+
+        # loop over each calendar year entry and extract data
+        for calendarYear in response:
+            calendarYearId = calendarYear["id"]
+            calendarYearName = calendarYear["name"]
+
+            # get available zones for calendar year
+            url = f"{baseUrl}/zones/?calendarId={calendarYearId}"
+            response = requests.request("GET", url)
```

Comment:
Please pass URL parameters using the params argument instead of string concatenation: https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls

Same is true for the following requests.

Reply:
okay, I switched to params even if I don't like it that much cause it tends to more loc needed and for me personally the readability is not that good. but anyways: you own the repo, you make the decision.